model-optimizer/mo/front/tf/graph_utils.py

```diff
@@ -185,3 +186,29 @@ def add_activation_function_after_node(graph: Graph, node: Node, activation_func
     else:
         raise Error('Unknown post-processing activation function "{}".'.format(activation_function))
     return activation_node
+
+
+def correct_roll_axes(roll: Node):
+    """
+    This function corrects axes of TF Roll node, if this node is before or after of TF (I)FFT operation.
+    :param roll: Node to correct axes.
+    :return: None
+    """
+    axes_source = roll.in_port(2).get_source()
+    axes_node = roll.in_port(2).get_source().node
+    axes_node_name = axes_node.soft_get('name', axes_node.id)
+
+    graph = roll.graph
+
+    less_node = create_op_with_const_inputs(graph, Less, {1: int64_array(0)}, {'name': axes_node_name + '/Less'})
+    mul_node = create_op_with_const_inputs(graph, Mul, {1: int64_array(-1)}, {'name': axes_node_name + '/Mul'})
+
+    roll.in_port(2).get_connection().set_destination(less_node.in_port(0))
+    less_node.out_port(0).connect(mul_node.in_port(0))
+
+    add_node = Add(graph, {}).create_node()
+    mul_node.out_port(0).connect(add_node.in_port(1))
+    axes_source.connect(add_node.in_port(0))
+    add_node.out_port(0).connect(roll.in_port(2))
+
+    rename_nodes([(axes_node, axes_node_name + '/axes_before_correction'), (add_node, axes_node_name)])
```

Comment:
You don't need to rename the original `axes_node` because the actual value is probably changed. Please, remove this line

Reply:
Fixed.